### ISSUE TITLE
 [get5_apistats.smx] [ERR] API request failed, HTTP status code: 500

### ISSUE THREAD

**Issue by YungGhost1377** [PhlexPlexico/get5-webapi#13]
Hi, can you tell me what's the matter?
```
VAC secure mode is activated.
GC Connection established for server version 1213, instance idx 1
hostname: 2x2#1
version : 1.37.7.4/13774 1213/8012 secure  [G:1:2295789]
udp/ip  : 127.0.1.1:27090  (public ip: 77.242.107.182)
os      :  Linux
type    :  community dedicated
map     : de_inferno
gotv[0]:  port 27077, delay 60.0s, rate 24.0
players : 0 humans, 1 bots (6/0 max) (not hibernating)

# userid name uniqueid connected ping loss state rate adr
# 2 "RESPAWN - GOTV" BOT active 24
#end
{
 "gamestate": 0,
 "available": 1,
 "plugin_version": "0.7.2-dev+510"
}
RecordSteamInterfaceCreation (PID 12696): STEAMHTTP_INTERFACE_VERSION002 / HTTP
[Get5] Loaded match config.
L 12/11/2020 - 22:11:22: [get5_apistats.smx] Failed to create logo directory: resource/flash/econ/tournaments/teams
Can't use cheat cvar sv_grenade_trajectory in multiplayer, unless the server has sv_cheats set to 1.
[Get5] Type !ready when your team is ready to veto.
[Get5] Type !ready when your team is ready to veto.
[Get5] An admin force ended the match.
L 12/11/2020 - 22:12:16: [get5_apistats.smx] [ERR] API request failed, HTTP status code: 500
L 12/11/2020 - 22:12:16: [get5_apistats.smx] [ERR] Response:
{
    "message": "Match is already finalized."
}
```

**Comment by PhlexPlexico** [comment]
I've run into this before. It's nothing game breaking, but when you end a match, it seems that the API wants to be called twice when ending, which causes the second finalization. I think this may have something to do with the API wanting to call it after it already updates the table to tell that the match is finished.

**Comment by PhlexPlexico** [comment]
As you can see here:
https://github.com/PhlexPlexico/G5API/blob/master/routes/matches/matchserver.js#L215

This is most likely called before any of the API calls that are brought back from the server - 
https://github.com/PhlexPlexico/get5-webapi/blob/master/scripting/get5_apistats.sp#L436

As an example. So it can be ignored as the game is trying to update data that has already been updated.

**Comment by YungGhost1377** [comment]
it started to happen after reinstalling os 🥺😢

**Comment by YungGhost1377** [comment]
Reinstall api to fix this problem ?

**Comment by PhlexPlexico** [comment]
No, nothing needs to be reinstalled. It can be ignored for now since it does not affect anything.

**Comment by YungGhost1377** [comment]
Okay, thanks a lot for your time.

**Comment by PhlexPlexico** [comment]
No problem.

For future reference, and a pull request could be made to fix this.

- On G5V, you hit cancel match
- It calls `/matches/{matchid}/cancel` on the API
- The API updates the database, and sets the proper values. The match becomes cancelled and can no longer be touched by the API (you get that error 500).
- After this call, the API then calls the game server to end the match (as seen [here](https://github.com/PhlexPlexico/G5API/blob/master/routes/matches/matchserver.js#L259)).
- The API then calls `OnSeriesResult` ([here](https://github.com/PhlexPlexico/get5-webapi/blob/master/scripting/get5_apistats.sp#L436)) but since the API has already cancelled the match, it produces the given error.

There's a few ways this can be adjusted. One is to edit the plugin, and allow a check to see if there is a series winner. If no winner exists, don't make the call. 

The second way, would be to adjust G5API, and call ending the match first before touching the database. This would remove the bug, and probably be the easiest fix.

Third would be to tell G5V to send the RCON Command to cancel the match, avoiding the API. Would definitely want to avoid this because it gets rid of the whole point of the `/cancel` route.
